google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java

```diff
@@ -16,4 +16,46 @@
 
 package com.google.cloud.storage.transfermanager;
 
-public class TransferManagerImpl {}
+import com.google.cloud.storage.BlobInfo;
+import com.google.common.collect.ImmutableList;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public final class TransferManagerImpl implements TransferManager {
+
+  private final TransferManagerConfig transferManagerConfig;
+  private final ExecutorService executor;
+
+  TransferManagerImpl(TransferManagerConfig transferManagerConfig) {
+    this.transferManagerConfig = transferManagerConfig;
+    this.executor = Executors.newFixedThreadPool(transferManagerConfig.getMaxWorkers());
+  }
+
+  @Override
+  public @NonNull UploadJob uploadFiles(List<Path> files, ParallelUploadConfig opts) {
+    List<Future<UploadResult>> uploadTasks = new ArrayList<>();
+    for (Path file : files) {
+      if (Files.isDirectory(file)) throw new IllegalStateException("Directories are not supported");
```

Comment:
just to clarify -- we plan to support Directory upload in the future?

Reply:
The idea is that the api will just accept file paths but we will "support" directories in the samples by showing the customers how to generate that list themselves from a directory.

Reply:
Are we planning to have `TransferManagerUtils` be public to consumers? If so, should we consider a method like `TransferManagerUtils.getFileListForDirectory(Path directory)` that does it for you?